lua/lspconfig/server_configurations/ast_grep.lua

```diff
@@ -0,0 +1,38 @@
+local util = require 'lspconfig.util'
+
+return {
+  default_config = {
+    cmd = { 'ast-grep', 'lsp' },
+    filetypes = { -- https://ast-grep.github.io/reference/languages.html
+      'c',
+      'cpp',
+      'rust',
+      'go',
+      'java',
+      'python',
+      'javascript',
+      'typescript',
+      'html',
+      'css',
+      'kotlin',
+      'dart',
+      'lua',
+    },
+    root_dir = util.root_pattern 'sgconfig.yaml',
```

Comment:
Hi @chrisgrieser, thanks for helping ast-grep! I wonder if we can also add `sgconfig.yml` here? It is also a popular yaml file extension.

Reply:
Sure. Not at my desk the next couple of days - probably quicker when you make a PR?